src/main/java/picard/arrays/MergePedIntoVcf.java

```diff
@@ -0,0 +1,260 @@
+            "<a href='https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3463112/#SEC2title'></a>" +
+            "<h4>Usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar MergePedIntoVcf \\<br />" +
+            "VCF=input.vcf \\<br />" +
+            "PED=zcall.output.ped \\<br />" +
+            "MAP=zcall.output.map \\<br />" +
+            "ZCALL_T_FILE=zcall.thresholds.7.txt \\<br />" +
+            "OUTPUT=output.vcf <br />" +
+            "</pre>";
+
+    private final Log log = Log.getInstance(MergePedIntoVcf.class);
+
+    private final ProgressLogger logger = new ProgressLogger(log, 10000);
+
+    @Argument(shortName = "VCF", doc = "The vcf containing the original autocall genotypes.")
+    public File ORIGINAL_VCF;
+
+    @Argument(shortName = "PED", doc = "PED file to be merged into VCF.")
+    public File PED_FILE;
+
+    @Argument(shortName = "MAP", doc = "MAP file for the PED file.")
+    public File MAP_FILE;
+
+    @Argument(shortName = "ZCALL_T_FILE", doc = "The zcall thresholds file.")
+    public File ZCALL_THRESHOLDS_FILE = null;
+
+    @Argument(shortName = "ZCALL_VERSION", doc = "The version of zcall used", optional = true)
+    public String ZCALL_VERSION = "UNKNOWN";
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "The output VCF file to write with merged genotype calls.")
+    public File OUTPUT;
+
+    private static HashMap<String, String[]> zCallThresholds = new HashMap<>();
+
+    @Override
+    protected int doWork() {
+        IOUtil.assertFileIsReadable(PED_FILE);
+        IOUtil.assertFileIsReadable(MAP_FILE);
+        IOUtil.assertFileIsReadable(ZCALL_THRESHOLDS_FILE);
+        IOUtil.assertFileIsWritable(OUTPUT);
+        try {
+            parseZCallThresholds();
+            final ZCallPedFile zCallPedFile = ZCallPedFile.fromFile(PED_FILE, MAP_FILE);
+            final VCFFileReader vcfFileReader = new VCFFileReader(ORIGINAL_VCF, false);
+            final VCFHeader header = vcfFileReader.getFileHeader();
+            if (header.getGenotypeSamples().size() > 1) {
+                throw new PicardException("MergePedIntoVCF only works with single-sample VCFs.");
+            }
+            addAdditionalHeaderFields(header);
+            writeVcf(vcfFileReader.iterator(), OUTPUT, vcfFileReader.getFileHeader().getSequenceDictionary(), header, zCallPedFile);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return 0;
+    }
+
+    private void parseZCallThresholds() {
+        final String notApplicable = "NA";
+        try (Stream<String> stream = Files.lines(ZCALL_THRESHOLDS_FILE.toPath())) {
+
+            stream.forEach(line -> {
+                final String[] tokens = line.split("\t");
+                if (tokens[1].equals(notApplicable) || tokens[2].equals(notApplicable)) {
+                    log.warn(String.format("Found a N/A threshold value - skipping. Tx: %s, Ty: %s", tokens[1], tokens[2]));
```

Comment:
This is not an unexpected condition (there are almost 4000 NAs in the psych chip thresholds file) - maybe a warn is too strong @yfarjoun ?  Debug?

Reply:
I meant to put a warning only when one is missing, but not the other, because in that case we are throwing away data (the other)

Reply:
It should be either both numeric or both N/A - otherwise that's an exception.